drivers/clock_control/clock_stm32_ll_mp2.c

```diff
@@ -46,10 +46,37 @@ static int stm32_clock_control_off(const struct device *dev, clock_control_subsy
 static int stm32_clock_control_get_subsys_rate(const struct device *dev,
 					       clock_control_subsys_t sub_system, uint32_t *rate)
 {
+	struct stm32_pclken *pclken = (struct stm32_pclken *)(sub_system);
+
 	ARG_UNUSED(dev);
-	ARG_UNUSED(sub_system);
-	ARG_UNUSED(rate);
-	return -ENOTSUP;
+
+	switch (pclken->bus) {
+	case STM32_CLOCK_PERIPH_USART1:
+		*rate = LL_RCC_GetUARTClockFreq(LL_RCC_USART1_CLKSOURCE);
+		break;
+	case STM32_CLOCK_PERIPH_USART2:
+	case STM32_CLOCK_PERIPH_UART4:
+		*rate = LL_RCC_GetUARTClockFreq(LL_RCC_UART24_CLKSOURCE);
+		break;
+	case STM32_CLOCK_PERIPH_USART3:
+	case STM32_CLOCK_PERIPH_UART5:
+		*rate = LL_RCC_GetUARTClockFreq(LL_RCC_USART35_CLKSOURCE);
+		break;
+	case STM32_CLOCK_PERIPH_USART6:
+		*rate = LL_RCC_GetUARTClockFreq(LL_RCC_USART6_CLKSOURCE);
+		break;
+	case STM32_CLOCK_PERIPH_UART7:
+	case STM32_CLOCK_PERIPH_UART8:
+		*rate = LL_RCC_GetUARTClockFreq(LL_RCC_UART78_CLKSOURCE);
+		break;
+	case STM32_CLOCK_PERIPH_UART9:
+		*rate = LL_RCC_GetUARTClockFreq(LL_RCC_UART9_CLKSOURCE);
```

Comment:
I have an issue with this implementation which will impose to update the function each time a new IP is added, not counting the suboptimized footprint.
I know this is the way it is done in MPU clock_control drivers, but this should be updated (on other MPU ports too) based on what is done on MCU  clock_control drivers.
This is a non negligible piece of work but highly required.

Reply:
This is correct.
This being said, thinking to it a bit more, this will require you to have the knowledge of the complete clock configuration, which you don't have today (unlike mp13) as this is done by the A35.
So let's keep it as is for now, and maybe we'll come back to it later on.